src/client/bedrock_client.rs

```diff
@@ -108,18 +108,18 @@ impl BedrockClient {
     pub async fn ask(
         &self,
         question: String,
+        template: Option<String>,
         context: Vec<String>,
         model: String,
     ) -> Result<String, ShellError> {
         let config = aws_config::load_from_env().await;
         let client = aws_sdk_bedrockruntime::Client::new(&config);
 
+        let tpl_value = template.unwrap_or( "Please answer this question: \\\"{}\\\". Using the following context: \\\"{}\\\"".to_string());
+        let mut rendered_tpl = tpl_value.replacen("{}", &*question, 1);
+        rendered_tpl = rendered_tpl.replacen("{}", &*context.join(" "), 1);
```

Comment:
This won't work, you'll just end up with the prompt without the question, since there are no "{}" to replace. You'd need to do something like: 

```
        let request_body = if let Some(tpl) = template {
            format!("{} - {}", tpl, question)
        } else {
            format!(
                "Please answer this question: \\\"{}\\\". Using the following context: \\\"{}\\\"",
                question,
                context.join(" ")
            )
        };

        let question_with_ctx = if !context.is_empty() {
            format!(
                "{}. Using the following context: \\\"{}\\\"",
                request_body,
                context.join(" ")
            )
        } else {
            request_body
        };
```

And when using a custom prompt do we want to support users passing in context as well, or should they be mutually exclusive?